pkg/k8s/kubectl.go

```diff
@@ -54,15 +78,40 @@ func (kctl *kubectl) ProxyScheme() string {
 	return "http"
 }
 
-func (kctl *kubectl) Version() ([3]int, error) {
-	var version [3]int
-	bytes, err := kctl.sh.CombinedOutput("kubectl", "version", "--client", "--short")
+func (kctl *kubectl) Version() (*VersionInfo, error) {
+	var versionStruct = &VersionInfo{}
```

Comment:
The suffix `Struct` makes me think that we are using [Hungarian Notation](https://en.wikipedia.org/wiki/Hungarian_notation), which shouldn't be the case. Maybe we could call the variable `versionInfo`?